source/common/stats/thread_local_store.h

```diff
@@ -16,6 +16,91 @@
 namespace Envoy {
 namespace Stats {
 
+/**
+ * A histogram that is stored in TLS and used to record values per thread. This holds two
+ * histograms, one to collect the values and other as backup that is used for merge process. The
+ * swap happens during the merge process.
+ */
+class ThreadLocalHistogram : public virtual Histogram {
+public:
+  virtual ~ThreadLocalHistogram() {}
+
+  /**
+   * Called in the beginning of merge process. Swaps the histogram used for collection so that we do
+   * not have to lock the histogram in high throughput TLS writes.
+   */
+  virtual void beginMerge() PURE;
+};
+
+/**
+ * Log Linear Histogram implementation per thread.
+ */
+class ThreadLocalHistogramImpl : public ThreadLocalHistogram, public MetricImpl {
+public:
+  ThreadLocalHistogramImpl(const std::string& name, Store& parent, std::string&& tag_extracted_name,
+                           std::vector<Tag>&& tags);
+
+  ~ThreadLocalHistogramImpl();
+  // Stats::Histogram
+  void recordValue(uint64_t value) override;
+
+  bool used() const override { return flags_ & Flags::Used; }
+  void beginMerge() override { current_active_ = 1 - current_active_; }
```

Comment:
Can you add a small comment here to the body of this function, just to point out this is a trick to swap current_active_ between 1 and 0.